db/migrate/20200415150049_set_provider_name_search_on_provider.rb

```diff
@@ -0,0 +1,12 @@
+class SetProviderNameSearchOnProvider < ActiveRecord::Migration[6.0]
+  def up
+    say_with_time "Adding normalised version of provider_name to provider_name_search" do
```

Comment:
As this is a once-off may be more appropriate to do via a temporary rake task
Keen to hear what others think ...

Reply:
Migration serves the purpose of snapshoting both
- structural changes 
- data manipulation


`20200415144710`_add_provider_name_search_to_provider.rb (structural changes)
`20200415150049`_set_provider_name_search_on_provider.rb (data manipulation)

when this is removed `20200415150049` and replace with a temporary rake task...
who will know
- when to execute it
- why you execute
- why database is broken
- why did we keep/remove temporary rake task

Reply:
I don't think rails migrations for data migration is the norm in Rail land @defong ...

From: https://www.ombulabs.com/blog/rails/data-migrations/three-useful-data-migrations-patterns-in-rails.html

**Pros**
- It's quick because you can ship a data migration in a few minutes
- You use Rails core logic that makes sure you don't run a migration more than once

**Cons**
- Migrations fall out of date very quickly and you don't have a quick way to find out
- You're not following Rails conventions: Files in db/migrate should migrate database structure, not database records

From: https://thoughtbot.com/blog/data-migrations-in-rails

"_Rails migrations should only be used for schema changes and not for actual data changes in the database._"

Reply:
That decision of not actual data changes in the migration was already defined by 20190227124613_initialize_provider_changed_at.rb

Reply:
Personally I'm fine with data migrations in rails migrations. A few things to note are:
- split it up from the structural change which is done
- down migration, probably not needed here as previous migration will remove the column
- if some one new starts the project they should be able to run all existing migrations without an error. This usually happens if there is existing data a migration needs is missing and causes a failure. Doesn't look like it will happen in this case.

Reply:
If we don't use data migrations, then I think we need to ensure our code is compatible with both versions of the data since there will be a time lag between when the migration runs and when the data will be updated. I don't mind moving towards that, but tbh I feel like it's more work to have code compatible with multiple versions in general.